pkg/client/status_test.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2024 Nutanix
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	nutanixtestclient "github.com/nutanix-cloud-native/cluster-api-provider-nutanix/test/helpers/prism-go-client/v3"
+)
+
+func Test_GetTaskStatus(t *testing.T) {
+	client, err := nutanixtestclient.NewTestClient()
```

Comment:
can we use "github.com/golang/mock/gomock" framework to mock interfaces and structs? eg: mockgen to create the wrappers for interfaces eg https://github.com/nutanix-core/k8s-agent/blob/main/k8s-agent/Makefile#L82 wherever possible

Reply:
I will refactor that once possible, but there is a bug in the SDK that prevents me from implementing the Service interface https://github.com/nutanix-cloud-native/prism-go-client/pull/150